### DISCUSSION TITLE
HTTPS/WSS

### DISCUSSION THREAD

**Discussion by nicandris** [Luligu/matterbridge#47] [Q&A] [answered]
Hey I've seen in the code that you have implemented ssl for frontend but there is no way to enable it currently. Is there a reason for having it disabled? I just changed the cert names and generated pem/key and it works fine

I'm trying to add matterbridge behind nginx proxy manager and it won't connect with https and ws mixed 🙈

**Comment by Luligu** [comment]
Hi,
They are disabled cause I didn't find a stable solution with the certificates.
What do you mean with mixed?

**Reply by nicandris** [reply]
```useWebSocket.js:11 Mixed Content: The page at 'https://matterbridge.blabla.website/' was loaded over HTTPS, but attempted to connect to the insecure WebSocket endpoint 'ws://192.168.178.209:8284/'. This request has been blocked; this endpoint must be available over WSS.```

for the certs you can just let the user use their own (with a couple of instructions on how to generate them)

**Comment by Luligu** [comment]
Hi,
thanks, now I get what you mean with mixed.
How did you create the certificate?
Maybe I can continue the certificate experiment in the future. 
As you see in the code it should have been used on both express and ws.
In Microsoft Edge I never made to make a reliable https connection.

**Comment by nicandris** [comment]
`openssl req -newkey rsa:2048 -new -nodes -x509 -days 3650 -keyout mykey.key -out mycert.pem` and I added them in frontend/certificates, so nothing much tbh 😅 

I kept it running for a while and no issues. the browser will of-course say its unsecure but thats expected (thats why im using nginx proxy manager)

**Comment by Luligu** [comment, marked as answer]
## [1.3.6] - 2024-06-28

### Changed

- [matterbridge]: Unified the http server port for the frontend and the WebSockerServer.
- [matterbridge]: Unified the https server port for the frontend and the WebSockerServer.
- [certificates]: The certificates for https connections are imported from the directory ~/.matterbridge/certs with these names: cert.pem, key.pem and ca.pem (optional). Use the -ssl command line parameter to activate https for both frontend and WebSocketServer.

### Fixed

- [matterbridge]: Fixed exports
- [matterbridgeDevice]: Fixed ElectricalEnergyMeasurement and ElectricalPowerMeasurement

**Comment by nicandris** [comment]
Worked like a charm! thank you 🙇 
![image](https://github.com/Luligu/matterbridge/assets/284798/8b9ac272-03b0-438a-884c-5f86ecb4789d)

(if any one wants to use this with docker(compose), you have to override the entrypoint with something like `command: ["matterbridge", "-bridge", "-docker", "-ssl"]`)

**Comment by Luligu** [comment]
Thanks.
Actually the not secure and all the warnings were really annoying me.
I also generate the ca.pem for my localhost certificate. Also windows is happy with that now and no more messages. All is secure like with the certificate you buy.

For the docker command, since now the possible and usefull flags are a lot, I will refactor the setting page of the frontend and add all command line parameter there. 
Matterbridge will start without any parameters using the saved configuration.
This will allow also to change from bridge mode to childbridge mode in docker without complex setup or override.
The command line will remain but will act like an override of the saved configuration.
Will not be tomorrow... cause designing the controls takes a lot of time.

**Comment by Luligu** [comment]
Done.
